schoolmatch.py

```diff
@@ -13,152 +20,171 @@
+#Will take a look at stop words next
+
+
+#read in all csvs and create relevant dataframes
+ospi_raw_pub_df = pd.read_csv('Washington_School_Directory_20171121.csv')
+ospi_raw_ind_df = pd.read_csv('ApprovedPrivateSchoolsList.csv')
+ospi_raw_ind_df.dropna(how='all', axis=1, inplace=True)
+ospi_raw_ind_df.dropna(how='all', axis=0, inplace=True)
+
+
+rs_raw_df = pd.read_csv('RSSchoolList2.csv', usecols=[0,1,2,3,4])
+rs_ind_df = rs_raw_df[rs_raw_df.Type != "Public"].copy()
+rs_pub_df = rs_raw_df[rs_raw_df.Type == "Public"].copy()
+
+
+
+#clean all of the city names as they are obviously variable among sources
+ospi_raw_pub_df.City = clean_title(ospi_raw_pub_df.City) 
+ospi_raw_ind_df.City = clean_title(ospi_raw_ind_df.City)
 
+rs_pub_df.City = clean_title(rs_pub_df.City)
+rs_ind_df.City = clean_title(rs_ind_df.City)
 
 
-RSSchoolList = RS_df.Name.tolist()
-RSSchoolListStrip = []
-RSSchoolListBones = []
 
-RSIndSchoolList = RS_df_NotPub.Name.tolist()
-RSIndSchoolListStrip = []
-RSIndSchoolListBones = []
+#Initialize empty variables to be used later
+rs_pub_list_trunc = []
+rs_ind_list_trunc = []
+ospi_pub_list_trunc = []
+ospi_ind_list_trunc = []
+ospi_pub_list_bycity_trunc = []
+ospi_ind_list_bycity_trunc = []
+ospi_pub_code = []
+ospi_pub_name = []
+ospi_pub_city = []
+ospi_ind_code = []
+ospi_ind_name = []
+ospi_ind_city = []
+all_pub_unmatched_df = []
+all_ind_unmatched_df = []
+
+
+#initialize vairbales of names
+rs_pub_list = rs_pub_df.Name.tolist()
+rs_ind_list = rs_ind_df.Name.tolist()
+ospi_pub_list = ospi_raw_pub_df.SchoolName.tolist()
+ospi_ind_list = ospi_raw_ind_df.SchoolName.tolist()
 
-RelevantOSPISchoolsBones = []
-RelevantOSPIIndSchoolsBones = []
 removablewords = ["Elementary", "Middle", "High", "School"]
 
+rs_pub_list_trunc = string_truncate(rs_pub_list,removablewords,rs_pub_list_trunc)  
+rs_ind_list_trunc = string_truncate(rs_ind_list,removablewords,rs_ind_list_trunc)
+ospi_pub_list_trunc = string_truncate(ospi_pub_list,removablewords, ospi_pub_list_trunc)
+ospi_ind_list_trunc = string_truncate(ospi_ind_list,removablewords, ospi_ind_list_trunc)
+
+ospi_pub_list_trunc = pd.Series(ospi_pub_list_trunc)
+ospi_raw_pub_df['TruncName'] = ospi_pub_list_trunc
+ospi_ind_list_trunc = pd.Series(ospi_ind_list_trunc)
+ospi_raw_ind_df['TruncName'] = ospi_ind_list_trunc
 
-#for z in RSSchoolList:
-#    y = str.strip(z)
-#    RSSchoolListStrip.append(y)
-#RSSchoolList = RSSchoolListStrip
 
-for z in RSSchoolList:
-    modWA = z
-    for w in removablewords:
-        modWA = modWA.replace(w, "")
-        modWA = modWA.strip()
-    RSSchoolListBones.append(modWA)
-    
-for z in RSIndSchoolList:
-    modWA = z
-    for w in removablewords:
-        modWA = modWA.replace(w, "")
-        modWA = modWA.strip()
-    RSIndSchoolListBones.append(modWA)    
-
-#Create a new dataframe to hold OSPI school info
-OSPIcode = []
-OSPIname = []
-OSPIcity = []
-
-OSPIIndcode = []
-OSPIIndname = []
-OSPIIndcity = []
 
 ### PUBLIC ###

```

Comment:
the code you have written for PUBLIC and NOT PUBLIC is nearly identical.  You should make this a single function that you call once for public and once for not public.  Maybe with a signature something like:
def matchSchools(rs_df, ospi_df):
You should only pass in the part of ospi_df that you are going to use, for example:
ospi_raw_pub_df[["SchoolCode","SchoolName","City","TruncName"]]
ospi_raw_ind_df[["School Bldg Number", "School Name", "City", "TruncName"]]

Reply:
Got it.